### ISSUE TITLE
kubectl fails with error Unable to connect to the server: net/http: TLS handshake timeout

### ISSUE THREAD

**Issue by apryiomka** [Azure/AKS#116]
As you can see by the UTC timestamp, the issue still persists and has not been resolved:
2018-01-10T18:17:43.6285407Z Unable to connect to the server: net/http: TLS handshake timeout
2018-01-10T18:17:43.7991087Z ##[error]Unable to connect to the server: net/http: TLS handshake timeout
2018-01-10T18:17:43.9339984Z ##[error]/opt/vsts/work/_temp/kubectlTask/1515608242301/kubectl failed with return code: 1

**Comment by apryiomka** [comment]
This is the original issue that has been close but has not been resolved: https://github.com/Azure/AKS/issues/112

**Comment by apryiomka** [comment]
BTW, az cli and kubectl fails with the same error:
PS C:\Users\user> kubectl get pods
Unable to connect to the server: net/http: TLS handshake timeout

**Comment by apryiomka** [comment]
BTW, az cli and kubectl fails with the same error:
PS C:\Users\user> kubectl get pods
Unable to connect to the server: net/http: TLS handshake timeout

**Comment by JackQuincy** [comment]
Sorry for not noticing this. Could you send me your sub id , resource group and resource name so I can look closer at what is not working properly in your api server?  Also what k8s version is this cluster?

**Comment by JackQuincy** [comment]
I ask about k8s version as we have been noticing issues with mostly 1.7.x clusters

**Comment by bottydim** [comment]
Same issue here in eastus
after aks scale the nodes were first not showing up and now:
kubectl get nodes
Unable to connect to the server: net/http: TLS handshake timeout

**Comment by apryiomka** [comment]
I am running V1.8.1. As of today, the issue still persists...

**Comment by fgauna12** [comment]
I was using a 1.7.x cluster and it was happening then. I upgraded it to 1.8.x and it was still happening. 
This was yesterday.

**Comment by jchauncey** [comment]
I am working on replicating this issue

**Comment by jchauncey** [comment]
```
BTW, az cli and kubectl fails with the same error:
PS C:\Users\user> kubectl get pods
Unable to connect to the server: net/http: TLS handshake timeout
```

@apryiomka what do you mean `az cli` fails with the same error?

**Comment by apryiomka** [comment]
Sorry, I created the original issue for VSTS deployment task that started to fail. My comment was to address az cli tool as well. If you connect to the Kube cluster using azure cli (az aks get-credentials --resource-group [rgroup name] --name [cluster name]), kubectl commands fail with the same error as VSTS deployment task. I don't know what VSTS task uses behind the scene, so i made this clarification:
```
2018-01-10T18:17:14.3331381Z ==============================================================================
2018-01-10T18:17:14.3787591Z Task         : Deploy to Kubernetes
2018-01-10T18:17:14.4243418Z Description  : Deploy, configure, update your Kubernetes cluster in Azure Container Service by running kubectl commands.
2018-01-10T18:17:14.4690084Z Version      : 0.1.7
2018-01-10T18:17:14.5131837Z Author       : Microsoft Corporation
2018-01-10T18:17:14.5593495Z Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=851275)
2018-01-10T18:17:14.6041909Z ==============================================================================
2018-01-10T18:17:24.6889939Z [command]/opt/vsts/work/_temp/kubectlTask/1515608242301/kubectl --kubeconfig /opt/vsts/work/_temp/kubectlTask/1515608242301/config apply -f /opt/vsts/work/r1/a/app/yaml/projectartifacts/deployment.yaml
2018-01-10T18:17:43.6285407Z Unable to connect to the server: net/http: TLS handshake timeout
2018-01-10T18:17:43.7991087Z ##[error]Unable to connect to the server: net/http: TLS handshake timeout
2018-01-10T18:17:43.9339984Z ##[error]/opt/vsts/work/_temp/kubectlTask/1515608242301/kubectl failed with return code: 1
2018-01-10T18:17:44.0682592Z ##[section]Finishing: Deploy Container to the Cluster
```
When i use azure cli, i get same error with kubectl commands, like kubectl get pods

**Comment by Karreg** [comment]
Same here.
As proposed in the initial bug report, I tried to upgrade k8s from 1.7 to 1.8, but the issue persists.
It seems to be related to westeurope for my part. A lot of feedback about issues with this region.

**Comment by Karreg** [comment]
I have created a cluster in westeurope, TLS Handshake issue.
I have created a cluster in eastus, no issue.
This is clearly related to westeurope...

**Comment by jchauncey** [comment]
So I just created a new cluster in west europe and was unable to reproduce this problem. So still looking to find out whats happening.

**Comment by jchauncey** [comment]
Ok Ive done a bit of digging and it seems that this is being caused by an etcd operator issue that we are working on fixing. Will update this issue once that is rolled out.

**Comment by chetanku** [comment]
Any updates on this?

**Comment by jchauncey** [comment]
@chetanku this issue was caused by a problem with the api server being in a crashloop iirc. Do you have a cluster that is currently up that is exhibiting this behavior? If so could you email me (jochaunc at microsoft.com) the details of the cluster (sub id, resource group, fqdn) and I will get on call to check it out.

**Comment by chetanku** [comment]
@jchauncey Thank you so much, I have sent you the details.

**Comment by marchanddesable** [comment]
same problem here, Azure AKS westeurope, since a few hours...

kubectl get all:
error: {batch  cronjobs} matches multiple kinds [batch/v1beta1, Kind=CronJob batch/v2alpha1, Kind=CronJob]

And also TLS Handshake issue if I try other commands...

Does it mean that we can't go to production with the current version of AKS ?
It was really what I needed, but this kind of issue scares me, I don't even know what to do with this cluster now... I see many ppl with the same problem, I don't want to pay for nothing ):

**Comment by scottlaplante** [comment]
@jchauncey I have a cluster that is up and exhibiting this problem. It is located in the Central US region. Would it be helpful to see another instance of this happening?

`$ kubectl get all
error: {batch  cronjobs} matches multiple kinds [batch/v1beta1, Kind=CronJob batch/v2alpha1, Kind=CronJob]`

`$ kubectl get jobs
Unable to connect to the server: net/http: TLS handshake timeout`

FWIW, I have no CronJobs (yet) - thinking about trying to upgrade to 1.8 to get them (would appreciate your thoughts on this).

**Comment by marchanddesable** [comment]
AKS is not ready, I have moved to Azure ACS with kubernetes.

**Comment by jchauncey** [comment]
I would recommend opening a support ticket so it can be routed to our on call. 

Thanks!

**Comment by jwilton-winshuttle** [comment]
We've had this happen on at least three separate occasions in which Azure support had to fix it.

It's been so common of an issue, with it taking days to resolve, we're going to have to ditch the managed service until it's out of preview.

**Comment by fgauna12** [comment]
I retried this today and it worked for me. East US. Following the quickstart on the docs. 1.8 cluster

**Comment by jiuchongxiao** [comment]
I have same issue.who know why and how to solve it?

**Comment by jwilton-winshuttle** [comment]
@jiuchongxiao 

Open a ticket with Azure support, they have to fix it. This has occurred at three separate times on multiple instances and they had to fix it every time. This is why we can't use it for production until it goes GA and may have to abandon AKS and move back to ACS for Dev/Test/Staging if it happens again.

**Comment by MovGP0** [comment]
I get the same error with Kubernetes version `1.10.3` on Docker version `18.05.0-ce-win68 (18263)`

**Comment by MovGP0** [comment]
I solved the issue by restarting docker. Also this issue should probably be reported in the [kubectl issue tracker](https://github.com/kubernetes/kubectl), since it not only affects Azure installations.

**Comment by azzaka** [comment]
I had the same issue on my cluster and it has been plagiung me for the past few months. My cluster is not on a Cloud Based provider, it's a Bare Metal Installation. I ran the command **curl -v8** which in turn showed my it wasn't a tls/cert issue (_really bad error message_.)

This lead me to look at other issues, such as the API server/gateway, nodes and so forth. It turns out that the the error --on my cluster-- was caused by mismatch API's. This was due to my adding a new node. After doing some digging I found a command that would allow me to see what API version/s my cluster was running and then guide me to an updated version. The command is probably known to most of you which is 'kubeadm upgrade plan'.

I updated my cluster using _'kubeadm upgrade plan'_. The command advised I could update the cluster to version _10.1.5_ or _10.1.11_, however I would need to updated Kubeadm first. I updated Kubeadm and then updated my kube components  across the other nodes. Once the nodes had been updated, I used the _kubeadm init join_ command to add all the nodes to the new cluster set. 
**NB** at this point I would like to say that none of my PODS dropped.

Everything had rejoined the cluster and now, I can now browse the logs across the cluster through all PODS.
